eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentEvents.scala

```diff
@@ -230,4 +230,6 @@ object PaymentFailure {
     }))
   }
 
-}
\ No newline at end of file
+}
+
+case class PathFindingExperimentMetrics(amount: MilliSatoshi, fees: MilliSatoshi, success: Boolean, duration: Long, timestamp: Long, experimentName: String)
```

Comment:
In AB testing we don't care about individual payments, we're only interested in the metrics averaged on many payments. If you want to manually try individual payments with different settings, that will be made easy with 0% experiments. But here the goal is to answer the question "I have two sets of parameters, which one should I use?" with minimal manual by just trying both on large enough samples.

Reply:
I think the issue in your argument lies in `on large enough samples`: we don't have that and won't have it in the foreseeable future. This discussion is similar to https://github.com/ACINQ/eclair/pull/1930#discussion_r704291804, let's continue on this thread to avoid splitting the discussion in two.